src/plugins/intel_gpu/src/graph/crop.cpp

```diff
@@ -21,23 +24,100 @@ layout crop_inst::calc_output_layout(crop_node const& node, kernel_impl_params c
     auto desc = impl_param.typed_desc<crop>();
     const auto& ref_in_sizes = desc->reference_input;
     const auto in_layout = impl_param.get_input_layout();
-    const auto& in_sizes = in_layout.get_tensor();
-    const auto& offsets = desc->offsets;
+    if (in_layout.is_static()) {
+        const auto& in_sizes = in_layout.get_tensor();
+        const auto& offsets = desc->offsets;
 
-    // Check for borders variant of crop.
-    if (ref_in_sizes.batch[0] < 0 || ref_in_sizes.feature[0] < 0 || ref_in_sizes.spatial[0] < 0 ||
-        ref_in_sizes.spatial[1] < 0 || ref_in_sizes.spatial[2] < 0) {
-        // Ignore not supported dimensions.
-        const auto rb_sizes = ref_in_sizes.negate().sub({0, 0, 0, 0, 0});
-        const auto lt_sizes = offsets.sub({0, 0, 0, 0, 0});
+        // Check for borders variant of crop.
+        if (ref_in_sizes.batch[0] < 0 || ref_in_sizes.feature[0] < 0 || ref_in_sizes.spatial[0] < 0 ||
+            ref_in_sizes.spatial[1] < 0 || ref_in_sizes.spatial[2] < 0) {
+            // Ignore not supported dimensions.
+            const auto rb_sizes = ref_in_sizes.negate().sub({0, 0, 0, 0, 0});
+            const auto lt_sizes = offsets.sub({0, 0, 0, 0, 0});
 
-        const auto out_sizes = in_sizes - (rb_sizes + lt_sizes);
+            const auto out_sizes = in_sizes - (rb_sizes + lt_sizes);
 
-        return layout({in_layout.data_type, in_layout.format, out_sizes});
+            return layout({in_layout.data_type, in_layout.format, out_sizes});
+        }
     }
     return layout({in_layout.data_type, in_layout.format, ref_in_sizes});
 }
 
+template<typename ShapeType>
+std::vector<layout> crop_inst::calc_output_layouts(const crop_node& /*node*/, const kernel_impl_params& impl_param) {
+    assert(static_cast<bool>(impl_param.desc->output_data_type) == false &&
+           "Output data type forcing is not supported for crop_node!");
+
+    auto desc = impl_param.typed_desc<crop>();
+    const auto in_layout = impl_param.input_layouts[0];
+    std::vector<ShapeType> output_shapes = {ShapeType()};
+    std::vector<ShapeType> input_shapes = {
+        impl_param.input_layouts[0].get<ShapeType>(),
+    };
+    for (size_t i = 1; i < impl_param.input_layouts.size(); ++i) {
+        input_shapes.push_back(impl_param.input_layouts[i].get<ShapeType>());
+    }
+
+    // TODO: calling shape_infer for all cropped outpus is redundant... Need to optimize.
+    if (impl_param.input_layouts.size() == 3) {
```

Comment:
Could you please modify this as 
1) pass the original op mode (e.g., split or variadic split) from the primitive creation 
2) use it here ?

I'm not meaning that passing ov::op itself, but create some new flags (enum)  presenting the original op whcih was converted to the crop.

Reply:
@yeonbok, It is good idea. but if crop is created internally in each node such as **_stride slice_** and **_rnn_**, which mode is selected from the original op mode? is it **_none_** enum?

Reply:
So they will go to else... :) 

Or can we add a handle to crop primitive ?
for example
crop_orig_node crop_mode() {
    if (inputs.size() == 2) {
         return crop::split;
     } else if (...)  {
          return crop_orig_node::variadic_split; 
     } else {
          return none; 
     }
}

What I would like is just make things more understandable, because current conditions might be hard to understand in the future.

Reply:
I understand what you want. I will fix it. thanks.

Reply:
Done